src/runtime/assemblymanager.cs

```diff
@@ -343,8 +353,17 @@ internal static void ScanAssembly(Assembly assembly)
             // A couple of things we want to do here: first, we want to
             // gather a list of all of the namespaces contributed to by
             // the assembly.
+            Type[] types = new Type[0];
+            try
+            {
+                types = assembly.IsDynamic ? assembly.GetTypes():assembly.GetExportedTypes();
```

Comment:
@dmitriyse  can you please add explanation for why you added check for `isDynamic`?

Reply:
should there be a test for this?

Reply:
According to the [documentation](https://msdn.microsoft.com/en-us/library/system.reflection.assembly.getexportedtypes%28v=vs.110%29.aspx?f=255&MSPPError=-2147217396), GetExportedTypes() is not supported on dynamic types and will throw *NotSupportedException*

Reply:
@Cronan good explanation! still maybe worth adding a comment about this. especially explaining the switch from GetTypes to GetExportedTypes for normal types.

Reply:
@dmitriyse @Cronan ok, I see this comment in the header of the PR:

Assembly.GetTypes replaced to Assembly.GetExportedTypes. So we are really skipping non-public types now.
Type.IsNested types are excluded from the names collection that is used for implicit assembly loading.